reference-manual/src/main/asciidoc/appendix.adoc

```diff
@@ -283,6 +284,158 @@ The URL used for phone home requests is
+The most obvious change is that many classes that used to be in the
+general `com.hazelcast.core` package are now in specific packages like
+`com.hazelcast.map` or `com.hazelcast.collection`.
+
+* We moved the serializable variants of `java.util.function` interfaces
+from `com.hazelcast.jet.function` to `com.hazelcast.function`:
+`BiConsumerEx`,`BiFunctionEx`,`BinaryOperatorEx`, `BiPredicateEx`,
+`ComparatorEx`, `ComparatorsEx`, `ConsumerEx`, `FunctionEx`, `Functions`,
+`PredicateEx`, `SupplierEx`, `ToDoubleFunctionEx`, `ToIntFunctionEx`,
+`ToLongFunctionEx`.
+
+* We updated/fixed the type parameters on several methods and classes, for
+example the `EntryProcessor`. See the
+https://docs.hazelcast.org/docs/latest/manual/html-single/#introducing-lambda-friendly-interfaces[relevant section]
+in the IMDG Migration Guide.
+
+* We changed the usage of the term "group" in configuration to "cluster".
+See the code snippet below for an example. This changes a Jet Command Line
+parameter as well (`-g/--groupName` renamed to `-n/--cluster-name`).
++
+[source]
+----
+include::{javasource}/migration/Migration.java[tag=config1]
+----
+
+* We moved `EventJournalConfig` from the top-level `Config` class to
+data structure-specific configs (`MapConfig`,`CacheConfig`).
++
+[source]
+----
+include::{javasource}/migration/Migration.java[tag=config2]
+----
+
+* Async methods that used to return `ICompletableFuture` now return
+`CompletionStage` (see the
+https://docs.hazelcast.org/docs/latest/manual/html-single/#removal-of-icompletablefuture[relevant section]
+in the IMDG Migration Guide).
+
+=== Jet API Changes
+
+We made multiple breaking changes in Jet's own APIs too:
+
+* We removed the Jet-specific interfaces `IMapJet`, `ICacheJet` and
+`IListJet` and now use IMDG's `IMap`, `ICache` and `IList` directly.
+
+* We renamed `Pipeline.drawFrom` to `Pipeline.readFrom` and
+`GeneralStage.drainTo` to `GeneralStage.writeTo`.
++
+[source]
+----
+include::{javasource}/migration/Migration.java[tag=pipeline1]
+----
+
+* We renamed `ContextFactory` to `ServiceFactory` and added support
+for instance-wide initialization. It now takes a `ProcessorSupplier.Context`
+instead of just `JetInstance`. We also added convenience methods in
+`ServiceFactories` to simplify constructing the common variants:
++
+[source]
+----
+include::{javasource}/migration/Migration.java[tag=serviceFactory1]
+----
+
+* We renamed `map/filter/flatMapUsingContext` to
+`map/filter/flatMapUsingService`.
++
+[source]
+----
+include::{javasource}/migration/Migration.java[tag=pipeline2]
+----
+
+* We removed the methods `withMaxPendingCallsPerProcessor(int)` and
+`withUnorderedAsyncResponses()` from `ServiceFactory`. These properties
+are relevant only in the context of asynchronous operations and were used
+in conjunction with `GeneralStage.mapUsingServiceAsync(...)`. In Jet 4.0
+the `GeneralStage.mapUsingServiceAsync(...)` method has a new variant with
+explicit parameters for the above settings.
++
```

Comment:
what about removal of filterUsingServiceAsync and flatMapUsingServiceAsync?

Reply:
I was under the impression that those weren't present before 4.0. Will include them.

Reply:
@jbartok I reworded to move the key words to the beginning, you can check it.

Reply:
@mtopolnik rewording looks good now

Reply:
@cangencer I've added info about removal of async filter & flatmap; pls. take a look, especially the code sample I've added for replacing flatMap, because I'm not sure that what I came up with is the right/best way to do it.